### PR TITLE
Viewing sms feature

### PR DESCRIPTION
**Pull request by DamianMizera**

### CODE REVIEW

**Comment by DamianMizera** [comment]
[Viewing SMS](https://trello.com/c/hpBEq7gB/4-viewing-sms)

File: src/UE/Application/Sms.cpp
[the remark this answers is not in the document]
Comment (SzymonSalamon) [reply src/UE/Application/Sms.cpp:1]: Deleted variable read and fucntions

File: src/UE/Application/Ports/BtsPort.cpp [hunk 1]
```diff
@@ -53,9 +53,8 @@ void BtsPort::handleMessage(BinaryMessage msg) {
       case common::MessageId::Sms: {
         auto action = reader.readNumber<std::uint8_t>();
         if (action == 0) {
-          const auto sms =
-              Sms{reader.readRemainingText(),      from, to, false, true,
-                  std::chrono::system_clock::now()};
+          const auto sms = Sms{reader.readRemainingText(), from, to, true,
```
Comment (SzymonSalamon) [inline src/UE/Application/Ports/BtsPort.cpp:56]: Change function to old one with read parametr.

File: src/UE/Application/States/SendingSmsState.cpp [hunk 2]
```diff
@@ -10,8 +10,8 @@ SendingSmsState::SendingSmsState(Context& context)
 void SendingSmsState::acceptButton() {
   std::string text = iSmsComposeMode.getSmsText();
   common::PhoneNumber toPhoneNumber = iSmsComposeMode.getPhoneNumber();
-  Sms sms = Sms{text, context.bts.getOwnPhoneNumber(), toPhoneNumber, false,
-                true, std::chrono::system_clock::now()};
+  Sms sms = Sms{text, context.bts.getOwnPhoneNumber(), toPhoneNumber, true,
```
Comment (SzymonSalamon) [inline src/UE/Application/States/SendingSmsState.cpp:13]: Change function to old one with read parametr.

File: src/UE/Application/ISmsDb.hpp [hunk 3]
```diff
@@ -1,5 +1,7 @@
 #pragma once
 
+#include <memory>
+
```
Comment (Bartosz-Slowik) [inline src/UE/Application/ISmsDb.hpp:4]:
```suggestion

```

File: src/UE/Application/Ports/UserPort.cpp [hunk 4]
```diff
@@ -32,11 +32,17 @@ void UserPort::showConnected() {
...
   menu.addSelectionListItem("Compose SMS", "");
-  menu.addSelectionListItem("View SMS", "");
+  if (smsDb.isUnreadSms()) {
+    menu.addSelectionListItem("View SMS (New!)", "");
+  } else {
+    menu.addSelectionListItem("View SMS", "");
+  }
   menu.addSelectionListItem("Dial", "");
 
+
```
Comment (Bartosz-Slowik) [inline src/UE/Application/Ports/UserPort.cpp:42]:
```suggestion

```

**Review by Bartosz-Slowik — APPROVED** [review]
I see you moved markasRead() to SmsDB i think its a good change but you can describe it in PR next time so it's easier to review.

**Review by SzymonSalamon — APPROVED** [review]

**Pull request merged**